manifests/fpm.pp

```diff
@@ -45,21 +45,21 @@
 #
 class php::fpm (
   $ensure               = $::php::ensure,
-  $service_ensure       = $::php::params::fpm_service_ensure,
-  $service_enable       = $::php::params::fpm_service_enable,
-  $service_name         = $::php::params::fpm_service_name,
-  $service_provider     = undef,
+  $service_ensure       = $::php::fpm_service_ensure,
+  $service_enable       = $::php::fpm_service_enable,
+  $service_name         = $::php::fpm_service_name,
+  $service_provider     = $::php::fpm_service_provider,
   $package              = "${::php::package_prefix}${::php::params::fpm_package_suffix}",
   $inifile              = $::php::params::fpm_inifile,
-  $settings             = {},
+  $settings             = $::php::real_settings,
   $global_pool_settings = {},
   $pools                = { 'www' => {} },
-  $log_owner            = $::php::params::fpm_user,
-  $log_group            = $::php::params::fpm_group,
-) inherits ::php::params {
+  $log_owner            = $::php::log_owner,
+  $log_group            = $::php::log_group,
+) {
 
   if $caller_module_name != $module_name {
-    warning('php::fpm is private')
+    fail('php::fpm is private')
```

Comment:
@igalic PR updated with the suggested rework.  `php::fpm::config` could do with a bunch of refactoring too, but that's a bigger task.